src/internal/auth/cf_auth_middleware.go

```diff
@@ -72,14 +72,9 @@ type promqlErrorBody struct {
 }
 
 func (m CFAuthMiddlewareProvider) Middleware(h http.Handler) http.Handler {
-	router := mux.NewRouter()
-
+	router := chi.NewRouter()
 	router.HandleFunc("/api/v1/read/{sourceID:.*}", func(w http.ResponseWriter, r *http.Request) {
-		sourceID, ok := mux.Vars(r)["sourceID"]
-		if !ok {
-			w.WriteHeader(http.StatusNotFound)
```

Comment:
Why remove the check here?

Reply:
because there's no equivalent to ok in the new code

Reply:
You would think. I think I added a check for "", It caused bigger problems, so I changed it to not exist instead.

Reply:
Maybe I lacked the return lol. Added the check back.

Reply:
Added back